PseudoGantry/lib/PseudoGantry/PseudoAxis.cpp

```diff
@@ -0,0 +1,121 @@
+        else return false;
+    }
+    else
+    {
+        // this is assigned 1 and not 0 because at this point our step_count_x is ((2 * STEPS) + 1)
+        // which means we are onto the next set of 8, and 17 would be the beginnig of that set
+        pseudo_axis->skip_counter = 1;
+        return false;
+    }
+}
+
+void isr_motor_pulse(PseudoAxis *pseudo_axis)
+{   
+    if (!toggle_encoder_output(pseudo_axis)) return;
+
+    bool direction = digitalRead(pseudo_axis->motor_dir_pin);
+    // Serial.print("direction:    "); Serial.println(direction);
+    // Serial.print("to skip:      "); Serial.println(pseudo_axis->changes_to_skip);
+
+    if (direction)
+    {
+        if (pseudo_axis->motor_position_current < pseudo_axis->axis_length_counts)
+        {
+            pseudo_axis->motor_position_current++;
+            // check if limit switch was pressed before this move
+            // checks whether the homing routine is being run
+            if (!pseudo_axis->ls_home.status)
+            { // if !PRESSED where PRESSED = 0
+                // sets limit switch to unpressed
+                pseudo_axis->ls_home.status = UNRESSED;
+                // using digital write so it can work on Due ot Uno, execution time ~ 3 microseconds
+                digitalWrite(pseudo_axis->ls_home.output_pin, UNRESSED);
+
+                // resets currrent position to 0
+                pseudo_axis->motor_position_current = 0;
+            }
+        }
+        else
+        {
+            // cant exceed max length - press limit switch
+            pseudo_axis->ls_far.status = PRESSED;
+
+            digitalWrite(pseudo_axis->ls_far.output_pin, PRESSED);
+        }
+    }
+    else if (!direction)
+    {
+        if (pseudo_axis->motor_position_current > 0)
+        {
+            pseudo_axis->motor_position_current--;
+
+            if (!pseudo_axis->ls_far.status)
+            { // if !PRESSED where PRESSED = 0
+                // sets limit switch to unpressed
+                pseudo_axis->ls_far.status = UNRESSED;
+
+                digitalWrite(pseudo_axis->ls_far.output_pin, UNRESSED);
+
+                // resets currrent position to max length of gentry
+                pseudo_axis->motor_position_current = pseudo_axis->axis_length_counts;
+            }
+        }
+        else
+        {
+            pseudo_axis->ls_home.status = PRESSED;
+            digitalWrite(pseudo_axis->ls_home.output_pin, PRESSED);
+        }
+    }
+    // dump_data(pseudo_axis);
+}
+
+void reset_pseudo_axis(PseudoAxis *pseudo_axis)
+{
+    pseudo_axis->skip_counter=0;
+    pseudo_axis->motor_position_current = pseudo_axis->motor_position_default;
+}
+
+void dump_data(PseudoAxis *pseudo_axis)
+{
+    if (Serial) {
+        String home_status;
+        String far_status;
+        Serial.print("Axis:              "); Serial.println(pseudo_axis->axis_name);
+        Serial.print("Axis length:       "); Serial.println(pseudo_axis->axis_length_counts);
```

Comment:
Can we add units to the print statement? ie. `"Axis length (counts): "` and "Motor position (..units...)"